internal/services/containers/kubernetes_cluster_node_pool_resource_test.go

```diff
@@ -2544,3 +2559,55 @@ resource "azurerm_kubernetes_cluster_node_pool" "test" {
 }
  `, data.Locations.Primary, data.RandomInteger)
 }
+
+func (KubernetesClusterNodePoolResource) sourceSnapshotId(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
```

Comment:
We cannot set this in tests - if this needs to be set in order for the test to pass then we should be looking at which resources are preventing deletion and why.